app/webpacker/components/Tickets/TicketCommentCreate.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Button, Modal } from 'semantic-ui-react';
+import { useMutation } from '@tanstack/react-query';
+import MarkdownEditor from '../wca/FormBuilder/input/MarkdownEditor';
+import useInputState from '../../lib/hooks/useInputState';
+import createComment from './api/createComment';
+import Loading from '../Requests/Loading';
+import Errored from '../Requests/Errored';
+
+export default function TicketCommentCreate({
+  open, onClose, ticketId, currentStakeholder, refetchComments,
+}) {
+  const [comment, setComment] = useInputState();
+
+  const {
+    mutate: createCommentMutation,
+    isPending,
+    isError,
+  } = useMutation({
+    mutationFn: createComment,
+    onSuccess: () => {
+      setComment('');
+      refetchComments();
```

Comment:
You shouldn't need to do this here. Your thinking probably comes from the old `useLoadedData` and `sync`, which was fine because there was no other way.

Now we're starting to use `tanstack-query` more and more, which has much better features and is more powerful. So you can do something you didn't do before: You can access the `QueryClient` (the thing that stores and basically "caches" backend API responses using query keys) and tell it to replace data directly.

Reply:
I will make a suggestion around the whole mutation to show you how this basically looks like. It is only a skeleton, and I will leave some gaps in the details so that you can fill them in yourself as an exercise to get familiar with the "new way" of doing things.